Loop/Managers/DoseMath.swift

```diff
@@ -0,0 +1,441 @@
+                correcting: min,
+                minTarget: HKQuantity(unit: unit, doubleValue: minGlucoseTargets.minValue),
+                units: units
+            )
+        } else if eventualGlucoseValue > eventualGlucoseTargets.maxValue,
+            let minCorrectionUnits = minCorrectionUnits, let correctingGlucose = correctingGlucose
+        {
+            return .aboveRange(
+                min: min,
+                correcting: correctingGlucose,
+                minTarget: HKQuantity(unit: unit, doubleValue: eventualGlucoseTargets.minValue),
+                units: minCorrectionUnits
+            )
+        } else {
+            return .inRange
+        }
+    }
+
+    /// Recommends a temporary basal rate to conform a glucose prediction timeline to a correction range
+    ///
+    /// Returns nil if the normal scheduled basal, or active temporary basal, is sufficient.
+    ///
+    /// - Parameters:
+    ///   - correctionRange: The schedule of correction ranges
+    ///   - date: The date at which the temp basal would be scheduled, defaults to now
+    ///   - suspendThreshold: A glucose value causing a recommendation of no insulin if any prediction falls below
+    ///   - sensitivity: The schedule of insulin sensitivities
+    ///   - model: The insulin absorption model
+    ///   - basalRates: The schedule of basal rates
+    ///   - maxBasalRate: The maximum allowed basal rate
+    ///   - lastTempBasal: The previously set temp basal
+    ///   - duration: The duration of the temporary basal
+    ///   - minimumProgrammableIncrementPerUnit: The smallest fraction of a unit supported in basal delivery
+    ///   - continuationInterval: The duration of time before an ongoing temp basal should be continued with a new command
+    /// - Returns: The recommended temporary basal rate and duration
+    func recommendedTempBasal(
+        to correctionRange: GlucoseRangeSchedule,
+        at date: Date = Date(),
+        suspendThreshold: HKQuantity?,
+        sensitivity: InsulinSensitivitySchedule,
+        model: InsulinModel,
+        basalRates: BasalRateSchedule,
+        maxBasalRate: Double,
+        lastTempBasal: DoseEntry?,
+        duration: TimeInterval = .minutes(30),
+        minimumProgrammableIncrementPerUnit: Double = 40,
+        continuationInterval: TimeInterval = .minutes(11)
+    ) -> TempBasalRecommendation? {
+        let correction = self.insulinCorrection(
+            to: correctionRange,
+            at: date,
+            suspendThreshold: suspendThreshold ?? correctionRange.minQuantity(at: date),
+            sensitivity: sensitivity.quantity(at: date),
+            model: model
+        )
+
+        let scheduledBasalRate = basalRates.value(at: date)
+        var maxBasalRate = maxBasalRate
+
+        // TODO: Allow `highBasalThreshold` to be a configurable setting
+        if case .aboveRange(min: let min, correcting: _, minTarget: let highBasalThreshold, units: _)? = correction,
```

Comment:
Does this actually work? (This seems like an odd dependency between this function and .insulinCorrection()). Depends on how swift enums work, when some of the params are _, like `correcting: _` and `units: _`.